kernel/CMakeLists.txt

```diff
@@ -117,8 +117,33 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
-endif()
 
+  set(final_heap_size ${CONFIG_HEAP_MEM_POOL_SIZE})
+
+  # Import all custom HEAP_MEM_POOL size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_ADD_SIZE_ ${DOTCONFIG} add_size_keys)
+
+  set(add_size_sum 0)
+
+  foreach(add_size ${add_size_keys})
+    math(EXPR add_size_sum "${add_size_sum} + ${${add_size}}")
+  endforeach()
+
+  if(CONFIG_HEAP_MEM_POOL_SIZE LESS "${add_size_sum}")
+    if(NOT CONFIG_HEAP_MEM_POOL_IGNORE_MIN)
+      set(final_heap_size "${add_size_sum}")
+    endif()
+    # Only warn if default value 0 has been modified
+    if(NOT CONFIG_HEAP_MEM_POOL_SIZE EQUAL 0)
+      message(WARNING "
+      CONFIG_HEAP_MEM_POOL_SIZE is less than requested minimum:
+        ${CONFIG_HEAP_MEM_POOL_SIZE} < ${add_size_sum}
+      Setting the system heap size to ${final_heap_size}")
```

Comment:
Can see this being useful for testing sample apps maybe but seems like it would get annoying pretty fast for an end application - thoughts @henrikbrixandersen ?

Reply:
I guess that's why @jhedberg created the `HEAP_MEM_POOL_IGNORE_MIN` setting.

https://github.com/zephyrproject-rtos/zephyr/blob/96ed1db313dfda872bf1b8637b8338e2e8578130/kernel/Kconfig#L853-L859

Reply:
actually it seems this warning may be printed even if `CONFIG_HEAP_MEM_POOL_IGNORE_MIN=y` because CONFIG_HEAP_MEM_POOL_SIZE is being compared against the calculated `add_size_sum` regardless of the ignore setting.

Reply:
Now updated to only print the warning if `IGNORE_MIN` was not set.